### PR TITLE
[SPARK-9440][MLlib] Add hyperparameters to LocalLDAModel save/load

### PR DESCRIPTION
**Pull request by feynmanliang**
@jkbradley @MechCoder 

Resolves blocking issue for SPARK-6793. Please review after #7705 is merged.

### CODE REVIEW

**Comment by feynmanliang** [comment]
Jenkins test this please

(just rebased onto master)

**Comment by feynmanliang** [comment]
Just rebased onto master so history is messed up

**Comment by feynmanliang** [comment]
Jenkins test this please

**Comment by jkbradley** [comment]
I'll review now

**Comment by jkbradley** [comment]
@feynmanliang Is the JIRA number correct?

File: mllib/src/main/scala/org/apache/spark/mllib/clustering/LDAModel.scala [hunk 1]
```diff
@@ -306,22 +307,29 @@ object LocalLDAModel extends Loader[LocalLDAModel] {
 
     val thisFormatVersion = "1.0"
 
-    val thisClassName = "org.apache.spark.mllib.clustering.LocalLDAModel"
+    val classNameV1_0 = "org.apache.spark.mllib.clustering.LocalLDAModel"
```
Comment (jkbradley) [inline mllib/src/main/scala/org/apache/spark/mllib/clustering/LDAModel.scala]: "thisClassName" is the standard we've used so far, so I'd keep it.

**Comment by jkbradley** [comment]
That's the only issue I see.  One comment: If we support more topics in the future, we may need a new model format which stores the concentration parameters in Parquet instead of JSON, but I think this is OK for now.

**Comment by jkbradley** [comment]
Oh, and feel free to fix the "thisClassName" issue in a follow-up PR if you want this one merged right away.  LGTM otherwise

File: mllib/src/main/scala/org/apache/spark/mllib/clustering/LDAModel.scala (hunk 1 above)
Comment (feynmanliang) [inline mllib/src/main/scala/org/apache/spark/mllib/clustering/LDAModel.scala]: OK

**Comment by feynmanliang** [comment]
I can add the fix; DistributedLDAModel needs it as well.

**Comment by feynmanliang** [comment]
JIRA issue was wrong; I've corrected it.

**Comment by jkbradley** [comment]
OK LGTM pending tests

**Comment by feynmanliang** [comment]
Jenkins retest this please

**Comment by jkbradley** [comment]
2 out of 3 tests passed, with the failure from an unrelated test.  Good enough for me.  Merging with master.

**Pull request closed without merging**